src/Fcf/Data/List.hs

```diff
@@ -242,6 +279,36 @@ type family Drop_ (n :: Nat) (xs :: [a]) :: [a] where
   Drop_ _ '[]       = '[]
   Drop_ n (x ': xs) = Drop_ (n TL.- 1) xs
 
+-- | Type-level list takeWhile.
+--
+-- === __Example__
+--
+-- >>> :kind! Eval (TakeWhile ((>=) 3) '[1, 2, 3, 4, 5])
+-- Eval (TakeWhile ((>=) 3) '[1, 2, 3, 4, 5]) :: [Nat]
+-- = '[1, 2, 3]
+data TakeWhile :: (a -> Exp Bool) -> [a] -> Exp [a]
+type instance Eval (TakeWhile p '[]) = '[]
+type instance Eval (TakeWhile p (x ': xs)) =
+  (If (Eval (p x))
+      (x ': Eval (TakeWhile p xs))
```

Comment:
There's still an `Eval` here

Reply:
This looks very much like the Find-function that has been there already and the Find also has Eval in there. I can see that Filter has a bit different way there and I'll follow that way. I will change also the DropWhile at the same time.